Change certificate field valid from and valid to parsing to use pal.

simosillankorva opened the pull request:
Requires this pull request to be merged to work: https://github.com/ARMmbed/mbed-client-pal/pull/43.

Commit:
Implement server certificate expiration check

Commit:
Remove extra class qualification from header

simosillankorva committed:
Change to use PAL TLS (#63)

* Change to use the PAL TLS API.
* Add +1 to cert and key len. Change init done to use 2 states.
* Add enum for init stages.
* Fix unittests to work with palTLS changes + update stubs.
* Add option to enable setting entropy callback with flag MBED_CLOUD_CLIENT_CUSTOM_MBEDTLS_ENTROPY.
* Update README, add unit test instructions.

simosillankorva committed:
Change certificate field parsing to use pal.

simosillankorva committed:
Merge branch 'mbed-cloud-client-rel1.2' into valid_from_valid_to_pal_parsing

simosillankorva committed:
Fix and add unittests. Replase mbedtls entropy function call with PAL entropy call.

kuggenhoffen reviewed (commented):
Looks good, need to rebase and check again though

JanneKiiskila commented:
rebuild

JanneKiiskila commented on mbed-client-mbedtls/m2mconnectionsecuritypimpl.h:
```diff
@@ -157,14 +152,12 @@ class M2MConnectionSecurityPimpl{
-    uint8_t                     _init_done;
-    palTLSConfHandle_t          _conf;
-    palTLSHandle_t              _ssl;
+    uint8_t                             _init_done;
```
tab instead of space here?

kuggenhoffen replied:
No, it's just indented to same level as the lines below

JanneKiiskila commented on source/m2mconnectionsecuritypimpl.cpp:
```diff
@@ -357,8 +320,10 @@ bool M2MConnectionSecurityPimpl::check_server_certificate_validity(const M2MSecu
+        return false;
+    }
 
     if (device_time < server_validfrom || device_time > server_validto) {
```
should we add one check that server_validto > server_validfrom? I.e. malformed cert, dates would be in potentially wrong order, could happen.

Commit:
Merge branch 'mbed-cloud-client-rel1.2' into valid_from_valid_to_pal_parsing

* mbed-cloud-client-rel1.2:
  Temporarily disable cert expiration check
  Change cert expiration API's signature
  Fix cert validity month offset
  Add server cert check and small refactoring
  Fix cetificate parsing

Commit:
Fixes to certificate validity handling

- Moved temporary skip of check to caller to make unittests pass
- Fix unittests

Commit:
Add check for valid certificate validity fields

simosillankorva committed:
Merge branch 'mbed-cloud-client-rel1.2' into valid_from_valid_to_pal_parsing

simosillankorva commented:
rebuild

simosillankorva commented:
rebuild
